Trip overview

aamare1001 opened the pull request:
Adds in the Trip Cards for the dashboard
![Trips Dashboard mockup](https://user-images.githubusercontent.com/26231761/218926669-4cf4a0b8-ad36-42fe-8951-9af2eb6cf5a5.png)


Currently is static, does not add multiple per trips for user(just the UI Component)

aamare1001 committed:
Groundwork for TripCard

Lays groundwork for TripCard Component
Needs Avatars & Days until trip(for small)

Also adds flex to other bottom tab screens

aamare1001 committed:
Added more components of the screen

Added Headers, Time until, padding.

Needs Avatars, support for amount of trips

aamare1001 committed:
Added Avatars to the Trip Card

Added Avatars to the Large Trip Card
Also allows for ui kitten ui compnents with the Application Provider in the app.tsz

Victoria-Williamson commented on app/app.tsx:
```diff
@@ -26,6 +26,8 @@ import { GoogleSignin } from "@react-native-google-signin/google-signin"
 import { WEB_CLIENT_ID, TWITTER_APP_ID, TWITTER_KEY, TWITTER_SECRET } from "@env"
 import { Settings } from "react-native-fbsdk-next"
```
Remove unused imports

Victoria-Williamson commented on app/components/TripCard.tsx:
```diff
@@ -1,10 +1,87 @@
+import { Icon } from "./Icon"
+import { colors, typography } from "../theme"
+import Divider from "./Divider"
+import { Text } from "./Text"
```
I beleive you can also run ``yarn lint`` to check these ^^. But I can make a pre-commit hook so it shows up there

Victoria-Williamson commented on app/components/TripCard.tsx:
```diff
@@ -1,10 +1,87 @@
+            <Text text={`${getTime(trip.duration.start)}`} preset="subheading" size="sm" />
+          )}
+
+          {/* <Icon icon="personSmall" style={{left:300}}/> */}
```
What was this for? Was this an old version of the avatars?

aamare1001 replied:
Yes, it's scrap thank u!

Victoria-Williamson commented on app/screens/TripHome.tsx:
```diff
@@ -1,26 +1,99 @@
+  const background = {
+    back: require("../../assets/images/gradientBg.png"),
+  }
+  const exTrips = [
```
I would rename these trips, and when u display the trips below, do trips.map() and set up a function to separate current and upcoming trips so when we connect this to the data response from backend ( which will come in chronological order ) we just need to set this variable up to store the fetch response.

Victoria-Williamson commented on types/auth.ts:
```diff
@@ -7,3 +7,15 @@ export interface User {
   allergies: Array<string>
 }
+
+export type Trip = {
```
Sorry I forgot to mention I think. Move this to another file called ``trip.ts``

aamare1001 committed:
Trips are now iterative

Resolves all comments
Trips are now iterative, given an array of trips, each trip is populated into current & previous sections
Images are now shown using uri's

aamare1001 committed:
Fixes Styles for Trip Cards
